### PR TITLE
Example app: Allow changing port, set default to 3001

### PR DESCRIPTION
**Pull request by nylen**
I'm running another app on 3000.

### CODE REVIEW

**Comment by nylen** [comment]
This also fixes the example `url.redirect` setting and clarifies that it is a redirect back to the app.

File: example/setting_example.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "client_id": "<your client id here>",
   "client_secret": "<your client secret here>",
+  "port": 3001,
   "url":  {
-    "redirect": "http://localhost/connect/res"
+    "redirect": "http://localhost:3001/connect/res"
```
Comment (retrofox) [inline example/setting_example.json]: I like the idea but I think we should keep this file as simple as possible. Also maybe we can use process var https://nodejs.org/docs/latest/api/process.html#process_process_argv. It's very common and useful.

Comment (nylen) [inline example/setting_example.json]: OK, how about this instead?

- Change the port to 3001
- Fix the `url.redirect` setting to include `:3001` and do not make it a configurable option

The problem with using `process.argv` to configure the port is that the `url.redirect` setting will be broken if you change to a different port.

Comment (retrofox) [inline example/setting_example.json]: mmm, again. This file should be as simple as possible. The port issue is related with your particular situation where you are running two or web app so you have to set different ports to can ... etc. But it doesn't imply that everybody will have same issue.
For this reason is that this file is named `setting_example.json` like a guide to create a new `setting.json` file:
https://github.com/Automattic/node-wpcom-oauth/blob/master/example/index.js#L14
https://github.com/Automattic/node-wpcom-oauth/tree/master/example#2---create-a-setting-json-file

Comment (nylen) [inline example/setting_example.json]: Creo que tenemos una falta de comunicación.  I will post a new, simpler commit.

**Comment by nylen** [comment]
New commit.  I kept `settings_example.json` simple but added the possibility to change the port setting.  I found some more things that need to be changed:

- Rename `setting` -> `settings` - this is better English usage
- Clean up wording in a couple of other places
- Add a list for optional settings.  Later we need to add `scope` here, this will allow for generating a WordPress.com global token:  https://developer.wordpress.com/docs/oauth2/

File: example/index.js [hunk 2]
```diff
@@ -8,16 +8,16 @@ var express = require('express');
 var WPOAuth = require('../');
 
 /**
- * Get setting data
+ * Get settings data
  */
 
-var setting = require('./setting.json');
+var settings = require('./settings.json');
```
Comment (timmyc) [inline example/index.js]: I made these exact same changes in my local copy :)

**Comment by timmyc** [comment]
Looks good to me.  As I commented above, s/setting/settings/ was a change I made locally too, makes a bit more sense.

And agree on the global token as well.

**Pull request merged**